src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -564,11 +565,11 @@ def shell_safe_json_parse(json_or_dict_string, preserve_order=False, strict=True
             logger.debug(ex)  # log the exception which could be a python dict parsing error.
 
             # Echo the JSON received by CLI
-            msg = "Failed to parse JSON: {}\nError detail: {}".format(json_or_dict_string, json_ex)
+            msg = "Failed to parse string as JSON:\n{}\nError detail: {}".format(json_or_dict_string, json_ex)
```

Comment:
`json_or_dict_string` may be a file path, so we don't decisively say it is a JSON string.